### PR TITLE
ObjStore empty list fix

### LINKED ISSUE #563: NatsObjStore.ListAsync hangs if bucket is empty
### Observed behavior

As per title, if an object store bucket has no entries, then calling ListAsync hangs and doesn't return.

### Expected behavior

Enumerator should return

### Server and client version

Client : .Net 2.3.3
Server : Various

### Host environment

_No response_

### Steps to reproduce

_No response_

### PR DESCRIPTION
**Pull request by Ivandemidov00**
Resolves #563

### CODE REVIEW

File: src/NATS.Client.JetStream/Internal/NatsJSOrderedPushConsumer.cs [hunk 1]
```diff
@@ -142,12 +142,13 @@ public async ValueTask DisposeAsync()
...
-        _commandChannel.Writer.TryComplete();
-        _msgChannel.Writer.TryComplete();
-
         await _consumerCreateTask;
+
+        _commandChannel.Writer.TryComplete();
         await _commandTask;
 
+        _msgChannel.Writer.TryComplete();
+
```
Comment (mtmk) [inline src/NATS.Client.JetStream/Internal/NatsJSOrderedPushConsumer.cs:151]: why did we change these? did you encounter an issue? (you're probably right to make these changes, I just can't remember my reasoning here so just curious)

**Review by mtmk — COMMENTED** [review]
Looking good 💯 just a quick question below.

File: src/NATS.Client.JetStream/Internal/NatsJSOrderedPushConsumer.cs (hunk 1 above)
Comment (Ivandemidov00) [reply src/NATS.Client.JetStream/Internal/NatsJSOrderedPushConsumer.cs:151]: Yes, I encountered some problem.  
Before exiting WatchAsync, if Enumerable is empty, call `pushConsumer.Init();`
The consumer is initialized and the same consumer is immediately deleted in the dispose method.  
The creation operation `_consumerCreateTask` has not yet had time to complete and it requires an active `_commandChannel`, but it is already closed. Based on this, I decided to change the order of stopping channel reading.

Comment (Ivandemidov00) [reply src/NATS.Client.JetStream/Internal/NatsJSOrderedPushConsumer.cs:151]: Otherwise, when Dispose we get `NatsJSApiException: consumer not found`.

Comment (mtmk) [reply src/NATS.Client.JetStream/Internal/NatsJSOrderedPushConsumer.cs:151]: awesome, thanks fixing that 💯 should we stick this as comment on top of the relevant lines do you think?

**Review by mtmk — APPROVED** [review]
LGTM thanks @Ivandemidov00

ps @Ivandemidov00 are you on slack.nats.io dotnet channel?

**Comment by mtmk** [comment]
ouch sorry @Ivandemidov00 I keep forgetting this finally we put a check in github. could you please sign your commits? cheers
image

**Comment by Ivandemidov00** [comment]
I signed the commits and moved them to the next branch https://github.com/nats-io/nats.net/pull/578.
I think we can close this request

**Pull request closed without merging**